lib/system.nim

```diff
@@ -1830,8 +1830,11 @@ when not defined(nimscript):
 when defined(nimV2):
   include system/arc
 
-import system/assertions
-export assertions
+when not defined(nimPreviewSlimSystem):
+  {.deprecated: """assertions is about to move out of system; use `-d:nimPreviewSlimSystem` and
+                import `std/assertions`.""".}
+  import std/assertions
+  export assertions
```

Comment:
review here

Reply:
This is a bad idea, users have no control over system.nim. Instead remove the import/export of assertions and patch the tests. Important packages can be compiled via `--useVersion:1.6` or similar.

Reply:
Ok, here is the first step: https://github.com/nim-lang/Nim/pull/19601 (support useVersion:1.6)

Reply:
Is something like this okay?

https://github.com/nim-lang/Nim/pull/19602